File: Binary Search/javascript/binary-search-driver.js
var binarySearch = require('./binary-search');
var fs = require('fs');

var file = fs.readFileSync('../random-numbers.txt').toString();
var numbers = file.split('\n').map(i => +i);

console.log(binarySearch(numbers, 4204, 0, numbers.length)); // 0
console.log(binarySearch(numbers, 2125353, 0, numbers.length)); // 201
console.log(binarySearch(numbers, 7638830, 0, numbers.length)); // 779
console.log(binarySearch(numbers, 9760609, 0, numbers.length)); // 1000
console.log(binarySearch(numbers, 9760608, 0, numbers.length)); // -1

File: Binary Search/python/binary-search-driver.py
from binarySearch import binarySearch

file = open('../random-numbers.txt')
array = []
for val in file.read().split():
    array.append(int(val))

print (binarySearch(array, 4204, 0, len(array)))  # 0
print (binarySearch(array, 2125353, 0, len(array)))  # 201
print (binarySearch(array, 7638830, 0, len(array)))  # 779
print (binarySearch(array, 9760609, 0, len(array)))  # 1000
print (binarySearch(array, 9760608, 0, len(array)))  # -1
